src/DotNetty.Buffers/UnpooledHeapByteBuffer.cs

```diff
@@ -217,6 +218,19 @@ public override long GetLong(int index)
             return this._GetLong(index);
         }
 
+        public override int GetMedium(int index)
+        {
+            this.EnsureAccessible();
+            return this._GetMedium(index).ToMediumInt();
```

Comment:
We need to call .ToMediumInt() because in fact _GetMedium returns Unsigned Int24. And in java code the _GetMedium is _GetUnsignedMedium in reality. I renamed it to _GeMedium because it was was out of the ordinary in the AbstractByteBuffer all abstracted functions are named as _GetShort, _GetInt, _GetLong and etc..